From db2e06244de0e39bb7ad29891429e43c7dccc2a0 Mon Sep 17 00:00:00 2001
From: Steve Kondik <steve@cyngn.com>
Date: Sat, 3 Jan 2015 05:13:26 -0800
Subject: [PATCH 022/105] power: Disable keyboard/button lights while
 dozing/dreaming

 * With hardkeys and doze mode enabled, entering suspend results in
   an epic battle over the lights. It's a bad situation. Disable
   them when we're sleepy.

Change-Id: I7f1fc35a1573717d1ea101a07c4171d6f66d1553
---
 .../core/java/com/android/server/power/PowerManagerService.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/power/PowerManagerService.java b/services/core/java/com/android/server/power/PowerManagerService.java
index 16362aacbbc..5a4507e3231 100644
--- a/services/core/java/com/android/server/power/PowerManagerService.java
+++ b/services/core/java/com/android/server/power/PowerManagerService.java
@@ -2151,7 +2151,7 @@ public final class PowerManagerService extends SystemService
                 final long nextProfileTimeout = getNextProfileTimeoutLocked(now);
 
                 mUserActivitySummary = 0;
-                if (mLastUserActivityTime >= mLastWakeTime) {
+                if (mWakefulness == WAKEFULNESS_AWAKE && mLastUserActivityTime >= mLastWakeTime) {
                     nextTimeout = mLastUserActivityTime
                             + screenOffTimeout - screenDimDuration;
                     if (now < nextTimeout) {
-- 
2.17.1

